libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -922,6 +1024,69 @@ string YulUtilFunctions::updateStorageValueFunction(Type const& _type, boost::op
 	});
 }
 
+string YulUtilFunctions::writeToMemoryFunction(Type const& _type)
+{
+	if (auto ref = dynamic_cast<ReferenceType const*>(&_type))
+	{
+		solAssert(
+			ref->location() == DataLocation::Memory,
+			"Can only update types with location memory."
+		);
+
+		return writeToMemoryFunction(*TypeProvider::uint256());
+	}
+
+	string const functionName =
+		string("update_memory_value_") +
+		_type.identifier();
+
+	return m_functionCollector->createFunction(functionName, [&] {
+		if (/*auto str = */dynamic_cast<StringLiteralType const*>(&_type))
```

Comment:
If you do not distinguish between source and target type, then I would argue that this case can never happen, because it would require a memory type with a name that can be written down to include a string literal type, which is impossible.

Reply:
fair point. So we can just remove it (or assert for it)